### ISSUE TITLE
Reset Password and New account emails are not sending but works when I press the Test button in mail settings

### ISSUE THREAD

**Issue by jozefbonnar** [pterodactyl/panel#2730]
**Background (please complete the following information):**
* Panel or Daemon: Panel
* Version of Panel:  1.1.1
* Server's OS: Ubuntu 18.04 Server
* Your Computer's OS & Browser: Chrome

**Describe the bug**
The bug is the reset password not sending the email to the user when on the https://panel.example.com/auth/password or when creating a new user account but works in https://panel.example.com/admin/settings/mail when I click the "Test" button
Please provide additional information too, depending on what you have issues with:
Panel: `php -v` (the php version in use).
PHP 7.2.31-1+ubuntu18.04.1+deb.sury.org+1 (cli) (built: May 14 2020 08:27:35) ( NTS )

**To Reproduce**
1. Go to the Mail settings fill in all the details
2. Click test to send the email to make sure it works
3. Then either create a new account or go to the reset password page and type in a email with an account linked to the panel already
4. See your email and see that the email never game for reset password or new account details but the test email came

**Expected behavior**
A email should come for the test email but not for the reset password or when an account is created

**Comment by notAreYouScared** [comment]
Restart the `pteroq` service.

Edit: 1.x recommends PHP 7.3+ preferably 7.4. Seems you're using 7.2 and that might be where the issue is occurring.

**Comment by Trini8ed** [comment]
I'm also experiencing this issue, I'm uncertain how Pterodactyl grabs the login credentials whether its form the .env file or the sql database. After doing some debugging I've found that if I set up in the .env file to **APP_ENVIRONMENT_ONLY=true** the login credentials that I forced to the correct values are loaded in by the pteroq worker and the reset password emails are sent out correctly.

 As soon as I go back to the database variables instead of using the app environment file it seems to grab invalid/old login credentials. I only noticed this after deleting my G-Suite app password and was setting the email password to its new value. If anyone can comment on where the password/email settings are stored in the sql database I can manually force change it and see if that resolves this, though this would only be the case if my assumption is correct that they are indeed grabbed from sql and not the local .env file.

**Update: After doing some digging I found the settings variables and they seem to be updating in the database to a new different key each time I hit save on the email settings page. So it must be an issue with PHP decoding the password hashes from the SQL database and submitting invalid credentials to login to G-Suite. That or a different section of code is used that differs if **APP_ENVIRONMENT_ONLY=true**  is set to true.

**Comment by jozefbonnar** [comment]
Thank you @Trini8ed and @AreYouRlyScared after setting the APP_ENVIRONMENT_ONLY=true and restarted pteroq service it seems to be working 😊

**Comment by Trini8ed** [comment]
I'm not sure if settings the APP_ENVIRONMENT_ONLY variable is something you want to do as this is a development testing property I believe. I was tinkering with it a bit more last night and get it working after I edited the composer.json and changed "required" "php=^7.2" to 7.4 and upgrading to PHP 7.4. It seemed after I did this I entered the correct credentials and hit save then all the reset emails seems to be sent out. Note the test button on the page breaks at this point everytime you come back and refresh and try to apply and it doesn't work. 

So upgrade to PHP 7.4 update the composer.json and run composer update. Then apply the correct settings hit save and never touch the email settings page again.
